Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/file_bugs.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (C) 2021-2022 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import re
+import sys
+
+from .command import Command
+from .pull_request import PullRequest
+
+from webkitbugspy import Tracker, radar
+from webkitcorepy import arguments, run, Terminal
+from webkitscmpy import local, log, remote
+from webkitpy.common.system.user import User
+
+from webkitbugspy.radar import Tracker as RadarTracker
+from webkitbugspy.bugzilla import Tracker as BugzillaTracker
+
+
+class FileBugs(Command):
+    name = 'file-bugs'
+    aliases = ['bugs']
+    help = 'File bugzilla issues for commits without one specified'
+
+    BUGZILLA_PLACEHOLDER = 'Need the bug URL (OOPS!).'
+    RADAR_PLACEHOLDER = 'Include a Radar link (OOPS!).'
+
+    @classmethod
+    def parser(cls, parser, loggers=None):
+        parser.add_argument(
+            '--remote', dest='remote', type=str, default=None,
+            help='Make a pull request against a specific remote',
+        )
+
+        # Probably want some more args to configure the behaviour around
+        # auto-editing commit messages etc
+
+    @classmethod
+    def parse_description(cls, lines):
+        # If there's less than 6 lines it can't really match the commit template
+        if len(lines) < 6:
+            return
+
+        # Check if there's something on the 3rd line (for a rdar:)
+        # If so, then start our search for the description one line
+        # later.
```

Comment:
The commit object centralizes this logic in the `issues` property. But more to the point of this function, I've found it's most reliable to use empty lines, commit messages are usually in the format:

```
<title>
<links>
                         // Empty line
Reviewed by...
                         // Empty line
<description>
                         // Empty line
<files changed>
```